atc/db/team.go

```diff
@@ -451,6 +456,14 @@ func (t *team) SavePipeline(
 		}
 	}
 
+	_, err = tx.Exec(`
```

Comment:
Actually now that I think of it I'm not sure if this is what we want. Deleting inactive jobs has come up before but I think it should be a separate issue/PR to discuss, since it's somewhat orthogonal to having the ability to rename jobs (right?)

Reply:
Currently, the user has no way of knowing that an inactive job exists unless they query the db and, to my limited knowledge, inactive jobs does not provide any value in the pipeline at the moment? 

I guess if we don't delete the inactive jobs then some issues might arise from this current implementation of name change. In the scenario that the update name already exists as an inactive job, then a workaround would need to be implemented.

My initial thoughts are to either:
1. prompt the user that this inactive job exists already in the db and get them to fix their config to just use the name field normally
or 
2. let it have the current behavior which makes the new job have the history of whatever the inactive job has
or 
3. some better solution?

What do you think?

Reply:
The only real value of keeping inactive jobs is it allows the user to comment out a job temporarily and bring it back later while preserving the history.

This is not a documented feature and has been surprising to some in the past, but I don't think it's necessary to remove it as part of this PR. Or did it help with the implementation in some way?

Reply:
I guess the use case for inactive jobs is slight different from my past experiences. 

Frequently when setting up a pipeline in the past, I would change the name of the jobs to 'wipe out' history for the jobs since there might have been errors when configuring the pipeline. This results in many inactive jobs saved to the db unknown to the maintainers. In that scenario, if someone was to update the name of a job to an inactive job name by accident, they would get an error.

Maybe an error should just be shown at this time and a new issue can be created to discuss about inactive jobs (the use cases, to have it or not, does anyone know it exists, etc)?

Reply:
Ahh, I see, so that's why you had to add it as part of this.

What if it deleted any existing inactive jobs with the desired name only when doing the rename?

Reply:
That sounds like an appealing solution. This way there won't be much changes to the current state of inactive jobs. Let me give this a try!